### PR TITLE
Added anonymous_field_data 

### PR DESCRIPTION
**Pull request by YoginiTayade**

### CODE REVIEW

**Comment by sudeeppr1998** [comment]
@YoginiTayade Please update mongodb lib into package json

File: src/dispatcher/dispatcher.js [hunk 1]
```diff
@@ -43,6 +45,10 @@ class Dispatcher {
             console.log("inside mysql")
```
Comment (gouravmore) [inline src/dispatcher/dispatcher.js:45]: Remove the unnecessarily added console.log

File: src/dispatcher/dispatcher.js [hunk 2]
```diff
@@ -43,6 +45,10 @@ class Dispatcher {
...
             this.logger.add(winston.transports.mysql, this.options);
             console.log('mysql transport enabled !!!');
+        }else if (this.options.dispatcher === 'mongodb') {
+          console.log("inside mongodb");
+          this.logger.add(winston.transports.mongodb, this.options);
+          console.log('mongodb transport enabled !!!');
         }else { // Log to console
             console.log("inside else")
```
Comment (gouravmore) [inline src/dispatcher/dispatcher.js:53]: remove this console.log

File: src/dispatcher/mongodb-dispatcher.js [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      message.events = anonymousEventsArr;
+
+      try {
+        const response = await fetch(process.env.UrlForAnonymousDataToALL, {
+          method: "POST",
+          body: JSON.stringify(message),
+          headers: { "Content-Type": "application/json" },
+        });
+
+        console.log(JSON.stringify(response));
```
Comment (sudeeppr1998) [inline src/dispatcher/mongodb-dispatcher.js]: @YoginiTayade Remove this console.log

File: src/dispatcher/mongodb-dispatcher.js [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+            syncts: message.syncts,
+          })
+        );
+      }
+    } catch (err) {
+      return callback(err);
+    }
+
+    if (
+      process.env.sendAnonymousDataToALL === "yes" &&
```
Comment (gouravmore) [inline src/dispatcher/mongodb-dispatcher.js]: process.env.sendAnonymousDataToALL === "yes" &&    process.env.UrlForAnonymousDataToALL
      
 replace to => 
 
process.env.UrlForAnonymousDataToALL &&  process.env.sendAnonymousDataToALL === "yes"

File: src/dispatcher/mongodb-dispatcher.js [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+            return false; // skip
+          }
+          return true;
+        })
+        .map((event) => {
+          event.context.uid = "anonymous";
+          event.context.cdata = event.context.cdata
+            .filter((cdataEle) => {
+              if (
+                cdataEle.type === "school_name" ||
```
Comment (gouravmore) [inline src/dispatcher/mongodb-dispatcher.js:102]: need to remove this EMIS specific code

File: src/dispatcher/mongodb-dispatcher.js [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                cdataEle.type === "class_studying_id" ||
+                cdataEle.type === "udise_code"
+              ) {
+                return false; // skip
+              }
+              return true;
+            })
+            .map((cdataEle) => {
+              if (cdataEle.type === "Buddy User") {
```
Comment (gouravmore) [inline src/dispatcher/mongodb-dispatcher.js:111]: need to remove this EMIS specific code

File: src/dispatcher/mongodb-dispatcher.js (hunk 5 above)
Comment (YoginiTayade) [reply src/dispatcher/mongodb-dispatcher.js:102]: we require this code

File: src/dispatcher/mongodb-dispatcher.js (hunk 6 above)
Comment (YoginiTayade) [reply src/dispatcher/mongodb-dispatcher.js:111]: we require this code

File: src/dispatcher/dispatcher.js (hunk 2 above)
Comment (YoginiTayade) [reply src/dispatcher/dispatcher.js:53]: we require this code

File: src/dispatcher/dispatcher.js (hunk 1 above)
Comment (YoginiTayade) [reply src/dispatcher/dispatcher.js:45]: we require this code

File: src/dispatcher/mongodb-dispatcher.js (hunk 4 above)
Comment (YoginiTayade) [reply src/dispatcher/mongodb-dispatcher.js]: resolved

File: src/dispatcher/mongodb-dispatcher.js (hunk 3 above)
Comment (YoginiTayade) [reply src/dispatcher/mongodb-dispatcher.js]: resolved

**Pull request merged**